specifyweb/backend/bulk_copy/bulk_copy.py

```
@@ -68,3 +73,56 @@ def collection_dispatch_bulk(request, model) -> HttpResponse:
+            )
+
+            entity = query.column_descriptions[0]["entity"]
+            pk_col = inspect(entity).primary_key[0]
+
+            ids = list(session.execute(
+                query.with_entities(pk_col).distinct()
+            ).scalars())
+
+    if len(ids) > 0:
+        # Delete selection of records
+        for id in ids:
```

Ruff (0.15.15)

[error] Variable `id` is shadowing a Python builtin

(A001)